File: argentum-django/api/tests/data/guests.py
from decimal import Decimal

from django.utils.dateparse import parse_datetime

from api.models.guest import Guest
from api.tests.utils.test_objects import TestObjects


class TestGuests(TestObjects):
    MODEL = Guest

    ROBY = MODEL(
        id=1,
        code='DEMO-00001',
        name='Roby Brushfield',
        mail='rbrushfield0@sohu.com',
        status='staff',
        checked_in=parse_datetime('2019-12-31T22:55:44Z'),
        card='567a',
        balance=Decimal('4.50'),
        bonus=Decimal('0.00')
    )

    SHEELAH = MODEL(
        id=2,
        code='DEMO-00002',
        name='Sheelah Arnault',
        mail='sarnault1@tuttocitta.it',
        status='pending',
        checked_in=None,
        card=None,
        balance=Decimal('7.00'),
        bonus=Decimal('2.00')
    )

    ALL = [ROBY, SHEELAH]

    # Models below are not stored in the DB, but rather used for POST deserialization testing.

    JOHANNA_MIN = MODEL(
        id=3,
        code='DEMO-00003',
        name='Johanna Doe',
        mail='jdoe2@tinypic.com',
        status='paid',
        checked_in=None,
        card=None,
        balance=Decimal('0.00'),
        bonus=Decimal('0.00')
    )

    JOHANNA_MAX = MODEL(
        id=3,
        code='DEMO-00003',
        name='Johanna Doe',
        mail='jdoe2@tinypic.com',
        status='paid',
        checked_in=parse_datetime('2019-12-31T23:13:52Z'),
        card='581a',
        balance=Decimal('81.62'),
        bonus=Decimal('69.11')
    )

    ROBY_PATCHED = MODEL(
        id=1,
        code='DEMO-00001',
        name='Toby Brushfield',
        mail='tbrushfield0@sohu.com',
        status='paid',
        checked_in=parse_datetime('2019-12-31T22:55:44Z'),
        card='567a',
        balance=Decimal('4.50'),
        bonus=Decimal('0.00')
    )

File: argentum-ui/src/app/common/rest-service/test-data/guests.ts
import {Guest} from '../../model/guest';
import {Statuses} from './statuses';

export namespace Guests {
  export const ROBY = new Guest(
    17010,
    'DEMO-00001',
    'Roby Brushfield',
    'rbrushfield0@sohu.com',
    Statuses.PAID,
    new Date('2019-12-31T22:55:44Z'),
    '567a',
    7.50,
    0.00
  );

  export const SHEELAH = new Guest(
    17020,
    'DEMO-00002',
    'Sheelah Arnault',
    'sarnault1@tuttocitta.it',
    Statuses.PENDING,
    null,
    null,
    6.00,
    2.00
  );

  export const ALL = [ROBY, SHEELAH];

  export const JOHANNA_MIN_REQUEST = new Guest(
    undefined,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    undefined,
    undefined,
    undefined,
    undefined,
    undefined
  );

  export const JOHANNA_MIN_RESPONSE = new Guest(
    17030,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    null,
    null,
    null,
    0.00,
    0.00
  );

  export const JOHANNA_MAX_REQUEST = new Guest(
    undefined,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    Statuses.PAID,
    new Date('2019-12-31T23:13:52Z'),
    '581a',
    undefined,
    undefined
  );

  export const JOHANNA_MAX_RESPONSE = new Guest(
    17031,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    Statuses.PAID,
    new Date('2019-12-31T23:13:52Z'),
    '581a',
    0.00,
    0.00
  );

  export const ROBY_PATCHED_REQUEST = new Guest(
    17011,
    undefined,
    undefined,
    undefined,
    undefined,
    new Date('2019-12-31T22:57:30Z'),
    undefined,
    undefined,
    undefined
  );

  export const ROBY_PATCHED_RESPONSE = new Guest(
    17011,
    'DEMO-00001',
    'Roby Brushfield',
    'rbrushfield0@sohu.com',
    Statuses.PAID,
    new Date('2019-12-31T22:57:30Z'),
    '567a',
    7.50,
    0.00
  );

  export const JOHANNA_LIST_CREATED_REQUEST = new Guest(
    undefined,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    undefined,
    undefined,
    undefined,
    undefined,
    undefined
  );

  export const JOHANNA_LIST_CREATED_RESPONSE = new Guest(
    17030,
    'DEMO-00003',
    'Johanna Doe',
    'jdoe2@tinypic.com',
    null,
    null,
    null,
    0.00,
    0.00
  );

  export const ROBY_LIST_PATCHED_REQUEST = new Guest(
    undefined,
    'DEMO-00001',
    'Toby Brushfield',
    'tbrushfield0@sohu.com',
    Statuses.PENDING,
    undefined,
    undefined,
    undefined,
    undefined
  );

  export const ROBY_LIST_PATCHED_RESPONSE = new Guest(
    17012,
    'DEMO-00001',
    'Toby Brushfield',
    'tbrushfield0@sohu.com',
    Statuses.PENDING,
    new Date('2019-12-31T22:55:44Z'),
    '567a',
    7.50,
    0.00
  );
}
